products/accredited-representation-management/technical-documentation/appoint-a-representative/poa_request_proposal.md

```diff
@@ -0,0 +1,160 @@
+
+#### The Model
+```ruby
+class PowerOfAttorneyRequest < ApplicationRecord
+  has_kms_key
+  has_encrypted :consent_data, key: :kms_key, **lockbox_options
+  blind_index :first_name, :last_name, :city, :state_code, :zip_code
+
+  belongs_to :user_account
+  belongs_to :requested_individual, class_name: 'AccreditedIndividual'
+  belongs_to :accepted_or_declined_by, class_name: 'AccreditedIndividual', optional: true
+
+  enum status: {
+    'pending' => 'pending',
+    'accepted' => 'accepted',
+    'declined' => 'declined',
+    'expired' => 'expired'
+  }
+
+  def accept!(portal_user)
+    terminate
+    self.status = 'accepted'
+    self.accepted_or_declined_by = portal_user
+
+    self.save
+  end
+
+  def decline!(portal_user, reason)
+    terminate
+    self.status = 'declined'
+    self.accepted_or_declined_by = portal_user
+    self.decline_reason = reason
+
+    self.save
+  end
+
+  def expire!
+    terminate
+    self.status = 'expired'
+
+    self.save
+  end
+
+  private
+
+  def terminate
+    self.first_name = nil
+    self.last_name = nil
+    self.city = nil
+    self.state_code = nil
+    self.zip_code = nil
+    self.consent_data = nil
+  end
+end
+```
+
+#### The Migration
+```ruby
+class CreatePowerOfAttorneyRequests < ActiveRecord::Migration[7.1]
+  def change
+    create_table :power_of_attorney_requests do |t|
+      t.references :user_account, foreign_key: true, type: :uuid
+      t.references :requested_individual, foreign_key: {to_table: 'accredited_individual'}, type: :uuid
+
+      t.string :first_name_bidx
+      t.string :last_name_bidx
+      t.string :city_bidx
+      t.string :state_bidx
+      t.string :zip_code_bidx
+
+      t.text :consent_data_ciphertext
+
+      t.string :status
+      t.datetime :expires_at
+      t.references :accepted_or_declined_by, foreign_key: {to_table: 'accredited_individual'}, type: :uuid
+      t.string :decline_reason
+
+      t.text :encrypted_kms_key
+      t.timestamps
+
+      t.index :first_name_bidx
+      t.index :last_name_bidx
+      t.index :city_bidx
+      t.index :state_bidx
+      t.index :zip_code_bidx
+    end
+  end
+end
+```
+
+#### Consent Data
+Consent data will be treated as jsonb and should only contain boolean values.
+```ruby
+{
+  address_change_authorization: true,
+  limitation_alcohol: true,
+  limitation_hiv: true,
+  limitation_s_c_a: true,
+  limitation_h_i_v: false
+}
+```
+
+### Short term persistence
+Due to the sensitive nature of this data, the various encrypted fields will only be stored for XX days. After XX days or a terminal status is reached (i.e. the request is accepted) only metadata will be kept in the database.
+
+#### Cleanup Job
+This would be a scheduled job and could run on any cadence. The first choice would be once per day to match `InProgressFormCleaner`.
+
+```ruby
+class PowerOfAttorneyRequestCleaner
```

Comment:
I have been curious about other approaches to expiration. Sidekiq is probably fine, but we could be somewhat nervous that Sidekiq is offline for "too long". `pg_cron` to poll from within the DB might be slick, but I'm not sure.

Reply:
I think Sidekiq should be fine. vets-api relies on Sidekiq for some many things, it autoscales, etc.

Also, I've never used `pg_cron` so I can't comment on it off the top of my head.